src/test/java/com/github/caldav4j/methods/MkCalendarTest.java

```diff
@@ -98,49 +93,19 @@ public void testCreateRemoveCalendarCollection() throws Exception{
 		String collectionPath = caldavCredential.home + caldavCredential.collection;
 
 		HttpMkCalendarMethod mk = new HttpMkCalendarMethod(collectionPath,
-				"My Display Name", "This is my Default Calendar",
-				"en");
-
-		HttpClient http = createHttpClient();
-		HttpHost hostConfig = createHostConfiguration();
-		HttpResponse response = http.execute(hostConfig, mk);
-
-		int statusCode = response.getStatusLine().getStatusCode();
-		// whatever successful status code the caldav server returns,
-		//   the base collection is created, and should be removed.
-		//   TODO CaldavFixture may handle it automagically
-		if ((statusCode < 300) && (statusCode >=200)) {
-			addedItems.add("");			
-		}
-		// if resource already exists, remove it on teardown
-
-		switch (statusCode) {
-		case CalDAVStatus.SC_METHOD_NOT_ALLOWED:
-		case CalDAVStatus.SC_FORBIDDEN:
-			addedItems.add("");			
-			break;
-		default:
-			break;
-		}
+				"My Display Name", "This is my Default Calendar", "en");
 
-		/// Test if the caldav server return the right status code
-		assertEquals("Status code for mk:", CalDAVStatus.SC_CREATED, statusCode);
+		fixture.executeMethod(CalDAVStatus.SC_CREATED, mk, true, null, true);
```

Comment:
This is probably fine but I'm not sure it will remove the resource on teardown even if that's already existing.
I'd say that it is not a real problem though as tests should be isolated.

Reply:
Oh, you're right, I forgot to add the call to `fixture.tearDown()`.

Reply:
@ioggstream, I'll revert the change of adding the `fixture.tearDown` as the test fails because the resource is being removed during the test itself.

Reply:
@ioggstream, I'll avoid adding the resource to the tearDown list as the test fails because the resource is being removed during the test itself.